src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentColor/DocumentColorEndpoint.cs

```diff
@@ -25,22 +23,23 @@ public DocumentColorEndpoint(ClientNotifierServiceBase languageServer)
 
             _languageServer = languageServer;
         }
-        public DocumentColorRegistrationOptions GetRegistrationOptions(ColorProviderCapability capability, ClientCapabilities clientCapabilities)
+
+        public RegistrationExtensionResult? GetRegistration(VSInternalClientCapabilities clientCapabilities)
         {
-            return new DocumentColorRegistrationOptions()
-            {
-                DocumentSelector = RazorDefaults.Selector,
-            };
+            const string ServerCapabilities = "colorProvider";
+            var options = new DocumentColorOptions();
+
+            return new RegistrationExtensionResult(ServerCapabilities, options);
         }
 
-        public async Task<Container<ColorInformation>> Handle(DocumentColorParams request, CancellationToken cancellationToken)
+        public async Task<ColorInformation[]> Handle(DocumentColorParamsBridge request, CancellationToken cancellationToken)
         {
             var delegatedRequest = await _languageServer.SendRequestAsync(LanguageServerConstants.RazorProvideHtmlDocumentColorEndpoint, request).ConfigureAwait(false);
-            var documentColors = await delegatedRequest.Returning<Container<ColorInformation>?>(cancellationToken).ConfigureAwait(false);
+            var documentColors = await delegatedRequest.Returning<ColorInformation[]>(cancellationToken).ConfigureAwait(false);
 
             if (documentColors is null)
             {
-                return EmptyDocumentColors;
+                return Array.Empty<ColorInformation>();
```

Comment:
Kind of nuts that the LSP spec doesn't state that this can return `null`. Probably an oversight on their side.

Reply:
Yeah, I had to double-check that one too. I guess the idea is that either you checked the document and there are no colors or there was an exception and you should throw?